From 29241cf9dda811297f0eb212fce963698dbf4a81 Mon Sep 17 00:00:00 2001
From: Matthias Maennich <maennich@google.com>
Date: Tue, 28 May 2019 12:38:16 +0100
Subject: [PATCH 560/830] ANDROID: Initial abi_gki_aarch64 definition
X-NVConfidentiality: public

abi_gki_aarch64.out contains the ABI definition corresponding to the
current sources and the configuration referred to in
build.config.gki.aarch64.

As part of the build.sh tooling it will be copied over into the
distribution for further inspection / analysis. See
https://android-review.googlesource.com/970737 for details on that
process.

This is the initial version of this definition to allow implementation
of workflows around it. It is not considered stable at this point. It is
expected that it will break with significant changes from either
upstream or with changes implemented specific for this tree. Automated
validation is supposed to catch differences between this definition and
the actual binary before they are introduced into the tree. At a later
stage this validation should be part of build.sh itself.

Bug: 133501930
Change-Id: I815940ee13037ad450547e0ab0786e37f0b83d9b
Signed-off-by: Matthias Maennich <maennich@google.com>
---
 build.config.gki.aarch64 | 24 +++++-------------------
 1 file changed, 5 insertions(+), 19 deletions(-)

diff --git a/build.config.gki.aarch64 b/build.config.gki.aarch64
index 653e82bf1965..352a6382b06c 100644
--- a/build.config.gki.aarch64
+++ b/build.config.gki.aarch64
@@ -1,19 +1,5 @@
-ARCH=arm64
-BRANCH=android-mainline
-CLANG_TRIPLE=aarch64-linux-gnu-
-CROSS_COMPILE=aarch64-linux-androidkernel-
-CC=clang
-DEFCONFIG=gki_defconfig
-EXTRA_CMDS=''
-KERNEL_DIR=common
-POST_DEFCONFIG_CMDS="check_defconfig"
-CLANG_PREBUILT_BIN=prebuilts-master/clang/host/linux-x86/clang-r353983c/bin
-BUILDTOOLS_PREBUILT_BIN=build/build-tools/path/linux-x86
-LINUX_GCC_CROSS_COMPILE_PREBUILTS_BIN=prebuilts/gcc/linux-x86/aarch64/aarch64-linux-android-4.9/bin
-FILES="
-arch/arm64/boot/Image.gz
-vmlinux
-System.map
-"
-STOP_SHIP_TRACEPRINTK=1
-BUILD_INITRAMFS=1
+. ${ROOT_DIR}/common/build.config.common
+. ${ROOT_DIR}/common/build.config.aarch64
+. ${ROOT_DIR}/common/build.config.gki
+
+ABI_DEFINITION=abi_gki_aarch64.xml
-- 
2.17.1

